source/actions3/actions/AMDetectorTriggerAction.cpp

```diff
@@ -57,29 +56,35 @@ AMDetectorTriggerAction::AMDetectorTriggerAction(const AMDetectorTriggerAction &
 #include "beamline/AMDetectorTriggerSource.h"
 void AMDetectorTriggerAction::startImplementation(){
 	// If you still don't have a detector, check the exposed detectors one last time.
-	//if(!detector_)
-	//	detector_ = AMBeamline::bl()->exposedDetectorByInfo(*(detectorTriggerInfo()->detectorInfo()));
 	if(!detector_ && AMBeamlineSupport::beamlineDetectorAPI())
 		detector_ = AMBeamlineSupport::beamlineDetectorAPI()->exposedDetectorByInfo(*(detectorTriggerInfo()->detectorInfo()));
 
 	if(!detector_) {
 		AMErrorMon::alert(this,
-						  AMDETECTORTRIGGERACTION_NO_VALID_DETECTOR,
-						  QString("There was an error trigger the detector '%1', because the detector was not found. Please report this problem to the Acquaman developers.").arg(detectorTriggerInfo()->name()));
+				  AMDETECTORTRIGGERACTION_NO_VALID_DETECTOR,
+				  QString("There was an error trigger the detector '%1', because the detector was not found. Please report this problem to the Acquaman developers.").arg(detectorTriggerInfo()->detectorInfo()->name()));
+		setFailed();
+		return;
+	}
+
```

Comment:
We might have to watch this one. It seems as if the isReadyForAcquisition check was something included in the SGMUpgrade branch, which might explain why that was an issue only seen on SGM, even though the code all appeared to be the same for the acquisition states on all the other beamlines.